From de2ee1989d5e1564eea5e71d76f5d1d87c612e8c Mon Sep 17 00:00:00 2001
From: anoopsjohn <anoopsamjohn@gmail.com>
Date: Tue, 2 Dec 2014 07:14:39 +0530
Subject: [PATCH 050/410] HBASE-12591 Ignore the count of mob compaction
 metrics when there is issue.(Jiajia Li)

Reason: Bug
Author: Li Jiajia
Ref: CDH-23659
---
 .../hadoop/hbase/mob/DefaultMobCompactor.java      |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/mob/DefaultMobCompactor.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/mob/DefaultMobCompactor.java
index c84ed82..d0c8175 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/mob/DefaultMobCompactor.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/mob/DefaultMobCompactor.java
@@ -171,14 +171,14 @@ public class DefaultMobCompactor extends DefaultCompactor {
                   KeyValue mobKv = KeyValueUtil.ensureKeyValue(cell);
                   mobKv.setSequenceId(kv.getSequenceId());
                   writer.append(mobKv);
+                  mobCompactedFromMobCellsCount++;
+                  mobCompactedFromMobCellsSize += cell.getValueLength();
                 } else {
                   // If the value of a file is empty, there might be issues when retrieving,
                   // directly write the cell to the store file, and leave it to be handled by the
                   // next compaction.
                   writer.append(kv);
                 }
-                mobCompactedFromMobCellsCount++;
-                mobCompactedFromMobCellsSize += cell.getValueLength();
               }
             } else {
               LOG.warn("The value format of the KeyValue " + kv
-- 
1.7.9.5

